src/Core/Content/Product/SalesChannel/Sorting/ProductSortingEntity.php

```diff
@@ -12,54 +12,22 @@ class ProductSortingEntity extends Entity
 {
     use EntityIdTrait;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $key;
+    protected string $key;
 
-    /**
-     * @var int
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $priority;
+    protected int $priority;
 
-    /**
-     * @var bool
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $active;
+    protected bool $active;
 
     /**
      * @var array<array{field: string, priority: int, order: ?string, naturalSorting: bool|int|null}>
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
      */
-    protected $fields;
+    protected array $fields;
```

Comment:
do we need to initialize this with an empty array? Otherwise it will throw an exception when you call createDalSortings without setting fields before

At least before there was a check when fields is not filled to fall back to an empty array